### ISSUE TITLE
terra-application-utility doc page reports issue about <ul> and <ol> must only directly contain <li>, <script> or <template> elements

### ISSUE THREAD

**Issue by bjankord** [cerner/terra-framework#561]
# Bug Report

## Description
terra-utility doc page reports issue about `<ul>` and `<ol>` must only directly contain `<li>`, `<script>` or `<template>` elements

## Steps to Reproduce

1. Navigate to https://engineering.cerner.com/terra-ui/#/components/terra-application-utility/application-utility/application-utility
2. Run the [axe chrome extension](https://chrome.google.com/webstore/detail/axe/lhdoppojpmngadmnindnejefpokejbdd?hl=en-US)
3. The following violation is reported: "`<ul>` and `<ol>` must only directly contain `<li>`, `<script>` or `<template>` elements"

## Additional Context / Screenshots
Need to review the markup and see how it is being rendering. There may be aria attributes that are throwing the role of the list items of or there may be other elements directly under the `<ul>` or `<ol>`

## Expected Behavior
No accessibility violations are reported when testing with axe-core

## Environment

* Component Name and Version: Terra Application Utility v2.5.0

**Comment by nramamurth** [comment]
Upon running the axe test and inspecting the DOM structure further, I noticed that the `readonly` list-item has the role `note` assigned to it. I confirmed that this is the issue by removing the role and running the axe test again.

According to the [documentation](https://www.w3.org/TR/html-aria/#docconformance), the roles that **MAY** be used for a `<li>` element are - menuitem, menuitemcheckbox, menuitemradio, option, none, presentation, radio, separator, tab, or treeitem. 

<img width="737" alt="Screen Shot 2019-03-27 at 12 28 03 PM" src="https://user-images.githubusercontent.com/10090859/55056320-deca8000-508b-11e9-9e2d-07352f4cbaa3.png">

**Reference:** 

https://github.com/cerner/terra-framework/blob/8bb100facebd04d23f88f0efda1356cb608e1b3a/packages/terra-application-utility/src/utility/_UtilityMenuItem.jsx#L159-L169

**Comment by nramamurth** [comment]
I could not navigate to the read-only item in the list using tabs / arrow keys. Is there a different way for the user to bring focus to the read-only item in the list? If the focus cannot be placed on the read-only item, I'd like to understand the purpose of the role attribute for the read-only item.

cc @neilpfeiffer @bjankord

**Comment by yuderekyu** [comment]
@nramamurth the read-only item by design is not interactive. So you cannot navigate or do actions upon it

**Comment by nramamurth** [comment]
@yuderekyu That's right. I'm just curious, if it cannot be navigated upon, then should it still provide accessibility? I do not think that the `role` attribute is being helpful to the screen readers because I am not able to access it.

**Comment by nramamurth** [comment]
@yuderekyu That's right but I am curious as to if we should provide accessibility
for non-interactive elements (which is the read-only item in this case). I am trying to understand the purpose of having the `role` attribute set to `note` for a non-interactive element.

@neilpfeiffer

**Comment by nramamurth** [comment]
I am wrong with the interpretation above, me thinks. I see this issue with a menu without a non-interactive element as well. 

<img width="961" alt="Screen Shot 2019-03-29 at 10 08 31 AM" src="https://user-images.githubusercontent.com/10090859/55209914-e06e8200-520a-11e9-938c-24f8f8c9e5b3.png">

**Comment by nramamurth** [comment]
@JakeLaCombe Should the non-interactive elements (read-only item here) support accessibility? I also wonder if `button` is a valid role for a `<li>` element.

**Comment by bjankord** [comment]
> I also wonder if button is a valid role for a `<li>` element.

I think  this is the root cause of the issue.
The accessibility tree is being rendering as follows with our current markup using `<ul>` / `<li>` and having `role="button"` on the list item:

```
ul
  button
  button
```

I don't think we need this marked up as an unordered list. I think we can switch the UL element to a DIV. We can also switch the LI elements to DIVs with the current role logic and then the accessibility tree should render as the following:

```
div
  button
  button
```
which would solve the that `<ul>` and `<ol>` must only directly contain `<li>`, `<script>` or `<template>` elements

**Comment by nramamurth** [comment]
I do not see the violations after changing the `ul` and `li` elements to `div`. 

<img width="959" alt="Screen Shot 2019-04-01 at 10 32 11 AM" src="https://user-images.githubusercontent.com/10090859/55304686-44db4c80-546a-11e9-942f-89cb3470bce6.png">

**Comment by bjankord** [comment]
@nramamurth Let's move forward with this change.
